resources/ansible/roles/uploader-metrics/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+---
+# The role is making the assumption that telegraf is installed in the base machine image.
+# This is to avoid locking errors with apt.
+
+- name: ensure target directory exists
+  file:
+    path: "{{ network_dashboard_repo_path }}"
+    state: directory
+
+- name: write private key to file
+  copy:
+    content: "{{ network_dashboard_private_key }}"
+    dest: "{{ network_dashboard_sk_path }}"
+    mode: '0600'
+
+- name: check if network dashboard repo already cloned
+  stat:
+    path: "{{ network_dashboard_repo_path }}/.git"
+  register: repo_stat
+
+- name: clone the network dashboard repo
+  git:
+    repo: "{{ network_dashboard_github_url }}"
+    dest: "{{ network_dashboard_repo_path }}"
+    version: "{{ network_dashboard_branch }}"
+    key_file: "{{ network_dashboard_sk_path }}"
+    accept_hostkey: yes
+  when: not repo_stat.stat.exists
+
+- name: update the network dashboard repo
+  git:
+    repo: "{{ network_dashboard_github_url }}"
+    dest: "{{ network_dashboard_repo_path }}"
+    version: "{{ network_dashboard_branch }}"
+    key_file: "{{ network_dashboard_sk_path }}"
+    accept_hostkey: yes
+    force: yes
+  when: repo_stat.stat.exists
+
+- name: ensure _telegraf can run safenode-manager without a password
+  lineinfile:
+    path: /etc/sudoers
+    state: present
+    line: "_telegraf ALL=NOPASSWD: /usr/local/bin/safenode-manager"
+    validate: "visudo -cf %s"
```

Comment:
This I think is not required for uploaders as it doesn't try to tap into safenode-manager.

Reply:
Ah cool, good spot, thanks!